src/Net/Socket.cpp

```diff
@@ -191,17 +179,12 @@ bool Socket::SendSync(const std::vector<uint8_t>& message, const bool incrementC
 
 void Socket::SendAsync(const std::vector<uint8_t>& message)
 {
-    bool first_in_queue = m_writeQueue.empty();
-    m_writeQueue.push(message);
-
-    if (!first_in_queue) {
+    if (!m_writeQueue.empty()) {
```

Comment:
This isn't correct. Since you aren't adding `message` to the queue, if something's already being written to the socket, we just drop `message`.